libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -510,3 +630,246 @@ void ContractLevelChecker::checkBaseABICompatibility(ContractDefinition const& _
 		);
 
 }
+
+void ContractLevelChecker::checkAmbiguousOverrides(ContractDefinition const& _contract) const
+{
+	vector<FunctionDefinition const*> contractFuncs = _contract.definedFunctions();
+
+	auto const resolvedBases = resolveBaseContracts(_contract);
+
+	FunctionSet collectedFunctions;
+
+	// Collect the most inherited functions of the direct base contracts
+	for (ContractDefinition const* contract: resolveBaseContracts(_contract))
+	{
+		FunctionSet baseSet = convertContainer<FunctionSet>(contract->definedFunctions());
+
+		for (FunctionDefinition const* function: getBaseFunctions(contract))
+		{
+			if (baseSet.cend() == baseSet.find(function))
```

Comment:
Why search for it? It is a set, you can just insert - if there a duplicates, the set will figure it out.

Reply:
You can even use `convertContainer<FunctionSet>` as above.

Reply:
Oh it's a multiset! Please rename the type alias...